sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1676,6 +1687,99 @@ def get_custom_materialization_type(name: str) -> t.Type[CustomMaterialization]:
     return strategy_type
 
 
+class EngineManagedStrategy(MaterializableStrategy):
+    def create(
+        self,
+        table_name: str,
+        model: Model,
+        is_table_deployable: bool,
+        render_kwargs: t.Dict[str, t.Any],
+        **kwargs: t.Any,
+    ) -> None:
+        is_snapshot_deployable: bool = kwargs["is_snapshot_deployable"]
+
+        if is_table_deployable and is_snapshot_deployable:
+            # We could deploy this to prod; create a proper managed table
+            logger.info("Creating managed table: %s", table_name)
+            self.adapter.create_managed_table(
+                table_name=table_name,
+                query=model.render_query_or_raise(**render_kwargs),
+                columns_to_types=model.columns_to_types,
+                partitioned_by=model.partitioned_by,
+                clustered_by=model.clustered_by,
+                table_properties=model.physical_properties,
+                table_description=model.description,
+                column_descriptions=model.column_descriptions,
+            )
+        elif not is_table_deployable:
+            # Only create the dev preview table as a normal table.
+            # For the main table, if the snapshot is cant be deployed to prod (eg upstream is forward-only) do nothing.
+            # Any downstream models that reference it will be updated to point to the dev preview table.
+            # If the user eventually tries to deploy it, the logic in insert() will see it doesnt exist and create it
+            super().create(
+                table_name=table_name,
+                model=model,
+                is_table_deployable=is_table_deployable,
+                render_kwargs=render_kwargs,
+                **kwargs,
+            )
+
+    def insert(
+        self,
+        table_name: str,
+        query_or_df: QueryOrDF,
+        model: Model,
+        is_first_insert: bool,
+        **kwargs: t.Any,
+    ) -> None:
+        deployability_index: DeployabilityIndex = kwargs["deployability_index"]
+        snapshot: Snapshot = kwargs["snapshot"]
+        is_snapshot_deployable = deployability_index.is_deployable(snapshot)
+
+        if is_first_insert and is_snapshot_deployable and not self.adapter.table_exists(table_name):
```

Comment:
Do we need the `self.adapter.table_exists(table_name)` check at all? I think we don't expect the table here and if it exists, then something went wrong and we should fail, shouldn't we?

Reply:
the table might already exist if the snapshot was deployable when `create()` was called

Reply:
Ah, excellent point. Ignore my suggestion then